include/message_filters/sync_policies/latest_time.h

```diff
@@ -0,0 +1,375 @@
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/**
+ * \brief Synchronizes up to 9 messages by their rates with upsampling via zero-order-hold.
+ *
+ * LatestTime policy synchronizes up to 9 incoming channels by the rates they are received.
+ * The callback with all the messages will be triggered whenever the fastest message is received.
+ * The slower messages will be repeated at the rate of the fastest message and will be updated
+ * whenever a new one is received. This is essentially an upsampling of slower messages using a
+ * zero-order hold (no interpolation).
+
+ * \section usage USAGE
+ * Example usage would be:
+\verbatim
+typedef LatestTime<sensor_msgs::CameraInfo, sensor_msgs::Image, sensor_msgs::Image> latest_policy;
+Synchronizer<latest_policy> sync_policies(latest_policy(), caminfo_sub, limage_sub, rimage_sub);
+sync_policies.registerCallback(callback);
+\endverbatim
+
+ * May also take an instance of a `rclcpp::Clock::SharedPtr` from `rclpp::Node::get_clock()` 
+ * to use the node's time source (e.g. sim time) as in:
+\verbatim
+typedef LatestTime<sensor_msgs::CameraInfo, sensor_msgs::Image, sensor_msgs::Image> latest_policy;
+Synchronizer<latest_policy> sync_policies(latest_policy(node->get_clock()), caminfo_sub, limage_sub, rimage_sub);
+sync_policies.registerCallback(callback);
+\endverbatim
+
+ * The callback is then of the form:
+\verbatim
+void callback(const sensor_msgs::CameraInfo::ConstPtr&, const sensor_msgs::Image::ConstPtr&, const sensor_msgs::Image::ConstPtr&);
+\endverbatim
+ *
+ */
+
+#ifndef MESSAGE_FILTERS__SYNC_LATEST_TIME_H_
+#define MESSAGE_FILTERS__SYNC_LATEST_TIME_H_
+
+#include <algorithm>
+#include <cassert>
+#include <deque>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <tuple>
+#include <vector>
```

Comment:
Given what is actually used in this header file, I think this whole block can be reduced to:

```
#include <algorithm>
#include <mutex>
#include <numeric>
#include <tuple>
#include <vector>
```

Reply:
done